### ISSUE TITLE
Latte & Used locations support?

### ISSUE THREAD

**Issue by Kcko** [gencer/HTML-Slim-CSS-SCSS-Class-Completion#9]
Could you please add support for Latte template system? Its similar to html or twig).
My config  "html-css-class-completion.includeGlobPattern": "**/*.{css,scss,sass,latte,html}", but not working :(

**Comment by gencer** [comment]
This is the first time I heard of Latte. I will try to implement basic support as soon as possible.

I've looked over few samples and its indeed like html/twig. It shouldn't be hard to implement.

I'm on it on this weekend.

**Comment by Kcko** [comment]
Its a primary template system for NETTE framework (https://latte.nette.org/en/) which wins 3 best PHP framework in the world for year 2015.

Really nothing hard, it has same structure like a HTML langguage with own macros (https://latte.nette.org/en/macros).

Thx for good job & thx for fast reply.

**Comment by Kcko** [comment]
For future, is there real chance to do autocompletion for SCSS? (autosuggestioning vars, functions, mixins etc?)

Your plugin is solving what Im still missing   in VSC (II've already reported it several times, without change/chance).

**Comment by gencer** [comment]
Currently SCSS autocompletes classes from CSS, SCSS, HTML and Slim files.

**For** _vars, functions and mixins_. **Yes**, This is possible and In fact I also need them too! But I will first add latte support.

**Comment by Kcko** [comment]
Great news! I'll look forward to it.

What is still basically missing in vsc is suggesting between 2 environments.

HTML -> CSS (SCSS, LESS ...)
and vice versa
CSS (SCSS, LESS ...) -> HTML

### this probably works (plugin to provide it).

main.scss { .test {color: red; } } // when im here, i will see .test2 selector from other css/scss file
rest.scss {.test2 {color: blue;} } / and vice versa

### some-file.html  ( plugin to provide it)

still we will see 
.test or test2 


## BUT

in  scss & css file i never see  classes or identifikators from (html / latte ...)  **REALLY MISS THIS FEATURE**

would it be possible?

Thx for reply and Im looking forward for latte extension :-)

**Comment by gencer** [comment]
> would it be possible?

Yes. It is possible. I will be working on it at this weekend. (Tomorrow and other day) If nothing goes wrong.

**Comment by gencer** [comment]
This feature has been added to master branch. I will provide an early beta extension for tests. Keep your eyes on here.

**Comment by gencer** [comment]
![gif](https://user-images.githubusercontent.com/1711576/52532825-c270bf00-2d3b-11e9-8399-376fd76c6de2.gif)


Extension released. Please update your vscode extensions.

**Comment by Kcko** [comment]
Hi, thx for latte support! Works great.

Suggesting between html(latte) and css (scss) works fairly well, it show but only classes (but no identificators "#"), could you add it?


Is possible via some settings turn on extended info about occurences what i´m looking for and what did  plugin find? Like here?  https://marketplace.visualstudio.com/items?itemName=solnurkarim.html-to-css-autocompletion It's better for an idea of what a class or identifier is, it will be very helfpul.

Thx for answer :-)
-

**Comment by gencer** [comment]
Yes, I can add `id` attributes. But let me ask this. When I look your link, I see that css autocompletion shows HTML files locations, where they are located at.

Do you want them? Or you want:


```
html:
<div id="here-is-an-id"></div>

scss/css:
#here-is... # autocomplete here.
```

**Comment by gencer** [comment]
Oh, I see, even i've added this feature already something is keeping this feature stopped. I will look into this now.

**Comment by Kcko** [comment]
>> Do you want them? Or you want:
But its the same or not? 


Yes, this is exactly what i  want. (screen when i used previous plugin)
![](http://bit.ly/2UVhNtf)

And in html / latte (when I searching class or id from some CSS or SCSS file) - it would be nice to show file where  it was found.
![](http://bit.ly/2UKRdCO)


## MIXED SCENARIO

![](http://bit.ly/2UKSjyq) 
![](http://bit.ly/2UVhVJf)  

When i create new id or class in HTML / LATTE and go to CSS / SCSS ur plugin will find it - COOL, but  how do I know if this is a class from CSS or HTML?


When we solved these things , plugin will be best on the MARKET and i will propage on Twitter and to our firm / colleagues :-)

**Comment by gencer** [comment]
Well, I was thinking this last week. I was thinking like `Go To Definition` feature. Sometimes, I also get lost due to this.

I will do my best to bring this to our extension.

I also think one more thing. I would like to separate `id` and `class` icons in autocomplete lists.

**Comment by Kcko** [comment]
>I also think one more thing. I would like to separate `id` and `class` icons in autocomplete lists.
Via some icon or with 2 separated lists?

When you  parse files you probably know file / line where  where the occurrence is located (when this information add to popup window which is shown after suggestion it will be timeless! :-)

**Comment by gencer** [comment]
I've implemented basic support for this. As you can see ID and Class has different icons. I will also separate them. When you press dot, you only see classes, when typing hash, you will see IDs.

![image](https://user-images.githubusercontent.com/1711576/52537958-ae4db180-2d7d-11e9-9a61-010230bce0f2.png)


This is how will it look like. A basic proposal.

**Comment by Kcko** [comment]
Good idea.

it could look like this: 
1) When i write some className or identifikator i will see everything in one list (it will be different by icon or some muted postfix/prefix (class) this-is-class or (id) this-is-id.
2) When i  specify selection with "." or "#" i get classname or id

And will you work previous point (extended info where class or id is located -> file/line or something like that).

Have a nice rest of the weekend

**Comment by gencer** [comment]
![image](https://user-images.githubusercontent.com/1711576/52575126-ba457c00-2e2e-11e9-88a8-caf45aacd481.png)

We're getting there...

**Comment by Kcko** [comment]
Looks fantastic!

**Comment by gencer** [comment]
@Kcko, have you tried what I say on other issue? Still same?

**Comment by gencer** [comment]
Better naming:

![image](https://user-images.githubusercontent.com/1711576/52575757-d0077100-2e2f-11e9-99b8-94eeb6cdc44b.png)

**Comment by Kcko** [comment]
> @Kcko, have you tried what I say on other issue? Still same?

Yes, you have an answer, look on video and my description, thx.

**Comment by gencer** [comment]
@Kcko please try this for "**usages**"

v1.5.1-beta: https://github.com/gencer/HTML-Slim-CSS-SCSS-Class-Completion/releases/tag/v1.5.1-beta

**Note**: This is still **BETA**. Need lot of work but it should give you the idea.
**Note 2:** Find usages only works in css/scss. So, in css/scss you should see your latte files.

**Comment by gencer** [comment]
There is a bug. When you create a new class in latte, usages are not listed. I've already fixed and will be available in next build.

**Comment by gencer** [comment]
Here is the fixed one: https://github.com/gencer/HTML-Slim-CSS-SCSS-Class-Completion/releases/tag/v1.5.2-beta

**Comment by Kcko** [comment]
I will try tomorrow, VSC offers 1.3.x version (http://bit.ly/2UTiXVX)

**Comment by gencer** [comment]
I did not publish, VSIX file is here: https://github.com/gencer/HTML-Slim-CSS-SCSS-Class-Completion/releases/tag/v1.5.2-beta

I will publish after some additional work.

**Comment by gencer** [comment]
Latte and find usages are released in marketplace with version of 1.5.3.
